changes/mapper.go

```diff
@@ -17,6 +18,7 @@ type KeyMapper interface {
 type TagMapper struct {
 	tags  []string
 	types map[reflect.Type]KeyIndexes
+	sync.RWMutex
```

Comment:
You're using a RWMutex yet not doing any read locking. I think you'll also want to add `mapper.RLock()` and `defer mapper.RUnlock()` calls in `KeyIndexes` before you perform the read.